src/lib.rs

```diff
@@ -3855,35 +3687,32 @@ fn get_i_opcode_and_encoding(tag: u8, word: &BitSlice<Lsb0, u8>) -> (Opcode, Ope
                         _ => { unreachable!() },
                     }
                 }
+            } else if word[13] {
+                match index {
+                    0 => TABLE4_23[table4_23_index as usize],
+                    1 => (Extr, I11),
+                    2 => (Purple, None),
+                    3 => (Shrp, I10),
+                    _ => { unreachable!() },
+                }
             } else {
-                if word[13] {
-                    match index {
-                        0 => TABLE4_23[table4_23_index as usize],
-                        1 => (Extr, I11),
-                        2 => (Purple, None),
-                        3 => (Shrp, I10),
-                        _ => { unreachable!() },
-                    }
-                } else {
-                    match index {
-                        0 => TABLE4_23[table4_23_index as usize],
-                        1 => (Extr_u, I11),
-                        2 => (Purple, None),
-                        3 => (Shrp, I10),
-                        _ => { unreachable!() },
-                    }
+                match index {
+                    0 => TABLE4_23[table4_23_index as usize],
+                    1 => (Extr_u, I11),
+                    2 => (Purple, None),
+                    3 => (Shrp, I10),
+                    _ => { unreachable!() },
                 }
             }
         },
-        6 => { (Purple, None) },
         7 => {
             // partial interpretation of table 4-16, `v_e == 1`
             if word[32] {
                 (Purple, None)
             } else {
                 // `Table 4-16 Multimedia and Variable Shift 1-bit Opcode Extensions`
                 // (`v_e == 0`, since `v_e == 1` parts of this table are all undefined)
-                const TABLE4_16: [&'static [(Opcode, OperandEncodingI); 64]; 4] = [
+                const TABLE4_16: [&[(Opcode, OperandEncodingI); 64]; 4] = [
```

Comment:
i imagine clippy linted this because `&'static` is the only valid lifetime in a `const`? i'd have never thought of this.

Reply:
Yeah, this is [redundant_static_lifetimes](https://rust-lang.github.io/rust-clippy/master/#redundant_static_lifetimes) taking effect.